ls1intum/Artemis
# Refactor more exam exercise group routes

### Checklist

### Motivation and Context

Breadcrumbs didn't work as intended for assessing exam exercises, since the routes were set up to redirect you to the course.
Closes #2998.
Closes #3004.
Closes #2702.

### Description

Now assessment is basically done "in the exam" route, so you can use the breadcrumbs. This also adds buttons to the Exercise Dashboards to make navigation a bit easier during assessment.

### Steps for Testing

(Note: "all exercise types" doesn't include quiz exercises for assessments.)

1. Test the assessment dashboard and exercise dashboards of all exercise types of a course
2. Test the assessment dashboard and exercise dashboards of all exercise types of an exam
3. Test "Scores", "Participations" and "Submissions" of exercises in an exam exercise group
4. Make sure assessing exercises of all types in an exam still works as intended (=> create a test exam with each exercise type, participate and assess)
5. Breadcrumbs should always display the exam title to click on when assessing exam exercises, all following segments do *not* necessarily need to work for now

It is easiest to cover everything by creating an exam, holding a test run (with assessment) and then participate with a few students and assess that.

(Also note that tutors often display IDs instead of titles, that is not a problem with this PR.)

### Screenshots

Example for breadcrumbs when assessing "programming exercise" (name of the exercise) of the "PreviousReturn" exam:
![grafik](https://user-images.githubusercontent.com/72132281/110557767-88a05b80-8141-11eb-9f51-42892aa4e812.png)

Buttons during assessment:
![grafik](https://user-images.githubusercontent.com/72132281/110799384-63176d00-827b-11eb-93cd-dc8f44dac0e2.png)
![grafik](https://user-images.githubusercontent.com/72132281/110799396-6579c700-827b-11eb-8055-4c4c1ab68024.png)

When there is no submission left to assess the following is displayed (except for programming and file-upload exercises):
![grafik](https://user-images.githubusercontent.com/72132281/110799467-7a565a80-827b-11eb-86e9-b0b5a1fbbd9e.png)

The following exercise-group buttons now leave you with the exam in the breadcrumbs:
![](https://user-images.githubusercontent.com/72132281/111079902-5f464d80-84fc-11eb-889e-2882671c31bf.png)
